appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/jwt/JwtTokenParser.java

```diff
@@ -81,9 +84,28 @@ public JwtTokenParser() {
         this(Optional.empty(), Optional.empty());
     }
     
-    public JsonWebTokenImpl parse(String bearerToken, String issuer, PublicKey publicKey) throws Exception {
-        SignedJWT signedJWT = SignedJWT.parse(bearerToken);
-
+     public void parse(String bearerToken) throws Exception {
+        rawToken = bearerToken;
+        signedJWT = SignedJWT.parse(rawToken);
+        
+        // MP-JWT 1.0 4.1 typ
+        if (!checkIsJWT(signedJWT.getHeader())) {
+            throw new IllegalStateException("Not JWT");
+        }
+    }
+    
+     public JsonWebTokenImpl parse(String bearerToken, String issuer, PublicKey signedBy) throws Exception {
+         try{
+             parse(bearerToken);
+             return verify(issuer, signedBy);
+         }catch(Exception e){
+             throw new IllegalStateException("", e);
```

Comment:
Out of curiosity, why throw a new exception here rather than just printing the stack trace of the caught exception with the same type? Could be a good reason for this, I'd be curious to know

Reply:
I thought it was somewhat strange, but the code comes from TokenParser class in MP TCK